src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -109,142 +172,113 @@ public void shouldNotCreateLocationIfConfidential() {
-            .location(location);
+        AccessibilityForDisabilities otherAccessibility = new AccessibilityForDisabilities().accessibility(OTHER_STRING);
+        AccessibilityForDisabilities newAccessibility = new AccessibilityForDisabilities().accessibility(NEW_STRING).details(NEW_STRING);
         newAccessibility.setIsConfidential(true);
+        Location location = generateExistingLocation();
+        location.setAccessibilities(Set.of(newAccessibility, otherAccessibility));
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        int dbSize = accessibilityService.findAll().size();
-        importService.createOrUpdateAccessibility(newAccessibility, location);
-
-        assertEquals(dbSize, accessibilityService.findAll().size());
+        assertEquals(1, accessibilityService.findAll().size());
+        assertEquals(OTHER_STRING, accessibilityService.findAll().get(0).getAccessibility());
     }
 
     @Test
     @Transactional
     public void shouldNotUpdateAccessibilityIfConfidential() {
+        AccessibilityForDisabilities otherAccessibility = new AccessibilityForDisabilities().accessibility(OTHER_STRING);
         Location location = generateExistingLocation();
-        AccessibilityForDisabilities accessibilityToBeUpdated = new AccessibilityForDisabilities().accessibility(EXISTING_STRING).details(EXISTING_STRING)
-            .location(location);
-        em.persist(accessibilityToBeUpdated);
-
-        AccessibilityForDisabilities otherAccessibility = new AccessibilityForDisabilities().accessibility(OTHER_STRING)
-            .location(location);
-        em.persist(otherAccessibility);
-        em.flush();
-        em.refresh(location);
-
-        AccessibilityForDisabilities newAccessibility = new AccessibilityForDisabilities().accessibility(EXISTING_STRING).details(NEW_STRING)
-            .location(location);
-        newAccessibility.setIsConfidential(true);
+        location.setAccessibilities(Set.of(otherAccessibility));
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
+        assertEquals(1, accessibilityService.findAll().size());
 
-        int dbSize = locationService.findAll().size();
-        AccessibilityForDisabilities updated = importService.createOrUpdateAccessibility(newAccessibility, location);
+        AccessibilityForDisabilities newAccessibility = new AccessibilityForDisabilities().accessibility(NEW_STRING).details(NEW_STRING);
+        newAccessibility.setIsConfidential(true);
+        location.setAccessibilities(Set.of(newAccessibility));
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, locationService.findAll().size());
-        assertNull(updated);
+        assertEquals(1, accessibilityService.findAll().size());
+        assertEquals(OTHER_STRING, accessibilityService.findAll().get(0).getAccessibility());
```

Comment:
In my opinion, this is misleading because the accessibility is outdated.

Reply:
same here. I agree, but we probably have to create a separate a task for it, as it's working the same way as before changing the ImportService.